packages/app-server-issuer/src/exchange-provider/ExchangeProviderManager.ts

```diff
@@ -72,27 +71,31 @@ export class ExchangeProviderManager {
         }),
       },
       this.context.ckitProvider,
-      this.providerAddress,
+      this.context.txSigner!.getAddress(),
     );
-    const unsignedTx = await builder.build();
-    const tx = await this.context.txSigner.seal(unsignedTx);
-    const hash = await this.context.ckitProvider.sendTransaction(tx);
-    await this.context.ckitProvider.waitForTransactionCommitted(hash);
+    try {
+      const unsignedTx = await builder.build();
+      const tx = await this.context.txSigner.seal(unsignedTx);
+      const hash = await this.context.ckitProvider.sendTransaction(tx);
+      await this.context.ckitProvider.waitForTransactionCommitted(hash);
+    } catch (e) {
+      console.log(e);
+    }
   }
 
   async initiateConfig(config: Config, context: ServerContext): Promise<void> {
     this.config = config;
-    this.providerAddress = helpers.generateSecp256k1Blake160Address(
-      hd.key.privateKeyToBlake160(this.context.ckitProvider.rpcUrl),
-    );
     this.context = context;
     this.initiated = true;
 
     await this.refreshCells();
   }
 
   exchangeAmount(sudtAmount: HexString): BigNumber {
-    return BigNumber.from(sudtAmount).mul(this.config.exchange.CKB).div(this.config.exchange.sUDT);
+    return BigNumber.from(sudtAmount)
+      .mul(this.config.exchange.CKB)
+      .div(this.config.exchange.sUDT)
+      .mul(BigNumber.from(10).pow(9));
```

Comment:
maybe there are some mistakes here,

1. we should multiply first and divide later to reduce the error caused by floating number

```
x / y = x1 / y1

# mul first and divide later
x = (x1 * y) / y1
```
2. all exposed RPC methods do not carry floating point values, for example CKB uses `shanon` as the unit, i think the `.mul(BigNumber.from(10).pow(9))` here is a mistake